### PR TITLE
UI: Keybind Menu

### PR DESCRIPTION
**Pull request by Alf21**
Still WIP as `ConsoleResult.Value` is `null` as well and there is no other way to access the console output or currently bound keys. Waiting for https://github.com/Facepunch/sbox-issues/issues/501 or https://github.com/Facepunch/sbox-issues/issues/172

### CODE REVIEW

File: code/ui/generalhud/menu/Menu.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
+using Sandbox.UI;
+using Sandbox.UI.Construct;
+
+namespace TTTReborn.UI
+{
+    public class Menu : Panel
```
Comment (mzegar) [inline code/ui/generalhud/menu/Menu.cs:6]: Do we want to make this name more specific, is this basically an options menu?

File: code/ui/generalhud/menu/Menu.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            get => _isShowing;
+            set
+            {
+                _isShowing = value;
+
+                SetClass("hide", !_isShowing);
+            }
+        }
+
+        private bool _isShowing = false;
```
Comment (mzegar) [inline code/ui/generalhud/menu/Menu.cs:21]: I wonder if we should have our own custom derived class that has this functionality, and each UI element can inherit from it, what do you think? We can add this as an issue.

File: code/ui/generalhud/menu/Menu.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+        public Menu()
+        {
+            Instance = this;
+
+            StyleSheet.Load("/ui/generalhud/menu/Menu.scss");
+
+            _keybindEntry = Add.KeybindEntry("Press a button");
+
+            SetClass("hide", true);
```
Comment (mzegar) [inline code/ui/generalhud/menu/Menu.cs:33]: Could technically use the property you made.

**Comment by Alf21** [comment]
Moved into https://github.com/TTTReborn/tttreborn/pull/177

**Pull request closed without merging**